core/storage/activity/gae_models_test.py

```diff
@@ -24,6 +24,7 @@
 class ActivityListModelTest(test_utils.GenericTestBase):
     """Tests the ActivityListModel class."""
 
+
```

Comment:
Remove this new line

Reply:
thanks @pranavsid98 it tried to add and it gives lint error for one line gap.

Reply:
Thats weird. @seanlip Do you have any idea?

Reply:
@apb7 should be able to advice. @GanitGenius it would be helpful if you could provide the exact text of the lint error here.

Reply:
ok, will do.
```
ckd@PC:~/Desktop/dev/oppia/oppia$ git push origin deprecate_old_classification_framework 
ref_list:
[GitRef(local_ref='refs/heads/deprecate_old_classification_framework', local_sha1='fe09b03b3a5cb90c5f729f79e363ec7b8a067580', remote_ref='refs/heads/deprecate_old_classification_framework', remote_sha1='783961291ecc1b26bbcc71f1fc0692624774bec1')]

Modified files in deprecate_old_classification_framework:
[FileDiff(status='M', name='core/storage/classifier/gae_models_test.py')]

Files to lint in deprecate_old_classification_framework:
['core/storage/classifier/gae_models_test.py']


Starting def-spacing checks
----------------------------------------

core/storage/classifier/gae_models_test.py:27:1: E302 expected 2 blank lines, found 1
1       E302 expected 2 blank lines, found 1
----------------------------------------

FAILED   Def spacing checks failed
```

i think this might be due to [this](https://github.com/oppia/oppia/blob/develop/scripts/pre_commit_linter.py#L679) where tox.ini contains E302 check. waiting for apurv to comfirm.

Reply:
@seanlip I tried this locally but did not get any lint error here. Also, initially there was a single line gap only between the class and the method and no linting error was encountered back then.  
@GanitGenius Could you please post the exact text of the lint error as suggested?  
Thanks!

Reply:
posting the full log:
```
ckd@PC:~/Desktop/dev/oppia/oppia$ git push origin deprecate_old_classification_framework 
ref_list:
[GitRef(local_ref='refs/heads/deprecate_old_classification_framework', local_sha1='fe09b03b3a5cb90c5f729f79e363ec7b8a067580', remote_ref='refs/heads/deprecate_old_classification_framework', remote_sha1='783961291ecc1b26bbcc71f1fc0692624774bec1')]

Modified files in deprecate_old_classification_framework:
[FileDiff(status='M', name='core/storage/classifier/gae_models_test.py')]

Files to lint in deprecate_old_classification_framework:
['core/storage/classifier/gae_models_test.py']


Starting def-spacing checks
----------------------------------------

core/storage/classifier/gae_models_test.py:27:1: E302 expected 2 blank lines, found 1
1       E302 expected 2 blank lines, found 1
----------------------------------------

FAILED   Def spacing checks failed

Starting import-order checks
----------------------------------------

----------------------------------------

Starting newline-at-EOF checks
----------------------------------------

----------------------------------------

(1 files checked, 0 errors found)
SUCCESS   Newline character checks passed
Starting linter...
Starting Javascript and Python Linting
----------------------------------------
Linting 1 Python files
Linting Python files 1 to 1...
There are no JavaScript files to lint.

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

Python linting finished.


----------------------------------------

SUCCESS   1 Python files linted (1.3 secs)

Starting Pattern Checks
----------------------------------------

----------------------------------------

(1 files checked, 0 errors found)
SUCCESS   Pattern checks passed
Push failed, please correct the linting issues above
error: failed to push some refs to 'git@github.com:GanitGenius/oppia.git'

```

Reply:
Commit changes:
```
ckd@PC:~/Desktop/dev/oppia/oppia$ git show fe09b03b3
commit fe09b03b3a5cb90c5f729f79e363ec7b8a067580 (HEAD -> deprecate_old_classification_framework)
Author: GanitGenius <anonims.ak@gmail.com>
Date:   Wed Feb 7 12:55:01 2018 +0530

    addressed review comment

diff --git a/core/storage/classifier/gae_models_test.py b/core/storage/classifier/gae_models_test.py
index e57267fa2..e50921f8b 100644
--- a/core/storage/classifier/gae_models_test.py
+++ b/core/storage/classifier/gae_models_test.py
@@ -24,7 +24,6 @@ import feconf
 (classifier_models,) = models.Registry.import_models(
     [models.NAMES.classifier])
 
-
 class ClassifierTrainingJobModelUnitTests(test_utils.GenericTestBase):
     """Test the ClassifierTrainingJobModel class."""

```
so earlier there was 2 line gap and after removing one line it gave that error.

Reply:
got it. i was modifing the wrong file '/activity/gae_models_test.py' instead of '/classifier/gae_models_test.py'. 
sorry for the trouble.

Reply:
@GanitGenius: Re E302, it checks for two lines above top-level functions and class definitions. E301 checks for single line gap between methods (https://github.com/PyCQA/pycodestyle/blob/master/pycodestyle.py#L306:L308).